compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/QueryTypeChecker.java

```diff
@@ -322,89 +323,149 @@ public BType resolveQueryType(SymbolEnv env, BLangExpression selectExp, BType ta
         }
     }
 
```

Comment:
For

```ballerina
type Foo record {
    Baz[] a;
};

type Bar record {
    boolean[]|float[] b;
};

type Baz record {
    string c;
};

type Quux record {
    string c;
    int d?;
};

type Qux record {
    string c;
    int d;
};

function transform(Foo foo) returns Bar => {
    b: from var _ in foo.a
        select {
           c: "str"
        }
};
```

there seems to be an extra error now.

```
a type compatible with mapping constructor expressions not found in type 'boolean'
```

Can we avoid this?

Reply:
This error is there even before the fix. This error comes when we try to type check the select clause againsts the expected member type of the query result.

Reply:
Will try to find a proper solution to remove this misleading diagnostic message.

Reply:
Since this error is expected logging is correct. The missing part was the error should be logged for all expected types. In this case `a type compatible with mapping constructor expressions not found in type 'float'` was missing. Added it in [#6a7b01](https://github.com/ballerina-platform/ballerina-lang/pull/40835/commits/6a7b01680f464c9fe8236d7d0f074d384c21e922).

Reply:
What are all the errors for this sample now?

Reply:
`ambiguous type '(Baz|Qux)'` will be the only error.